src/structures/member.ts

```diff
@@ -106,13 +110,64 @@ export async function createMemberStruct(
 
   // User was never cached before
   member.guilds.set(guildId, {
-    nick: nick,
-    roles: roles,
+    nick: props.nick,
+    roles: props.roles,
     joinedAt: Date.parse(joinedAt),
     premiumSince: premiumSince ? Date.parse(premiumSince) : undefined,
-    deaf: deaf,
-    mute: mute,
+    deaf: props.deaf,
+    mute: props.mute,
   });
 
-  return member as Member;
+  return member as MemberStruct;
```

Comment:
Avoid using type assertions and try to use type casting instead.
```suggestion
  return member;
```